fish-rust/src/common.rs

```diff
@@ -1985,6 +1995,100 @@ pub fn get_by_sorted_name<T: Named>(name: &wstr, vals: &'static [T]) -> Option<&
     }
 }
 
+/// Takes ownership of a variable and `Deref`s/`DerefMut`s into a projection of that variable.
+///
+/// Can be used as a workaround for the lack of `MutexGuard::map()` to return a `MutexGuard`
+/// exposing only a variable of the Mutex-owned object.
+pub struct Projection<T, V, F1, F2>
```

Comment:
`Projection` isn't used anywhere and I don't understand what it does; please remove it.

Reply:
You reviewed this before I finished rebasing and squashing everything. `Projection<T>` is used now (to refactor the changes that were added to `ScopeGuard` to make it support `scoped_push()` that drastically increased its complexity and made it unreadable) and I have more uses for it in the short-term.